addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -1837,131 +1837,146 @@ registry.TextHighlight = publicWidget.Widget.extend({
     selector: '#wrapwrap',
     disabledInEditableMode: false,
 
+    /**
+     * @constructor
+     */
+    init() {
+        this._super(...arguments);
+        // We need to adapt the text highlights on resize (E.g. custom fonts
+        // loading, layout option changes, window resized...), mainly to take in
+        // consideration the rendered line breaks in text nodes... But after
+        // every adjustment, the `ResizeObserver` will unfortunately immediately
+        // notify a size change once new highlight items are observed leading to
+        // an infinite loop. To avoid that, we use a lock map (`observerLock`)
+        // to block the callback on this first notification for observed items.
+        this.observerLock = new Map();
+        this.resizeObserver = new window.ResizeObserver(entries => {
```

Comment:
Do you actually need to declare this in the `init` instead of the start as before? It's ok but in general, using start for public widgets is more conventional.